rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1664,14 +1664,14 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 
 func Create_SubstitutionPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
 	inputs:
-		tradeState TradeState (1..1) <"The original trade to be modified.">
+		tradeState TradeState (1..1) <"The original trade to be for substitution of collateral.">
 		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new trade with new collateral.">
 		newPayout Payout(1..1) <"New payout to subtitute for the original payout. The payout define the product and other contractual attributes.">
-		priceQuantity PriceQuantity (1..*) <"The price and quantity of the substituted product. ">
+		priceQuantity PriceQuantity (1..*) <"The price and quantity of the substituted product.">
 	output: 
 		instruction PrimitiveInstruction (1..1)
 
-	condition ContractualProduct: <"Only a contractual product can be rolled.">
+	condition ContractualProduct: <"Only a repo contractual product can substitute collateral.">
```

Comment:
This is not correct, substitutions apply to securities lending trades too. We need to make sure this function works for both markets. At the moment it still closes the original trade - I know this is internal to the CDM but it still sets the original trade state to closedState which is ambiguous for the calling application. For this I believe we need 2 functions - one which closes the trade and creates a new one, and another which just replaces the collateral without closing the trade.